This is the melange bug that needs to be fixed first:
```
2023/10/13 02:58:33 [DEBUG] GET https://packages.wolfi.dev/bootstrap/stage3/x86_64/APKINDEX.tar.gz
Error: unable to build graph:
php-8.3-8.3_rc3-r1: unable to resolve dependency ${{package.name}}-config: could not find package, alias or a package that provides ${{package.name}}-config in indexes
time="2023-10-13T02:58:35Z" level=fatal msg="error during command execution: unable to build graph:\nphp-8.3-8.3_rc3-r1: unable to resolve dependency ${{package.name}}-config: could not find package, alias or a package that provides ${{package.name}}-config in indexes"
Error: Process completed with exit code 1.
```

--- a/php-8.3.yaml
+++ b/php-8.3.yaml
@@ -1,7 +1,7 @@
 package:
   name: php-8.3
   version: 8.3_rc3
-  epoch: 0
+  epoch: 1
   description: "the PHP programming language"
   copyright:
     - license: PHP-3.01
@@ -11,6 +11,7 @@ package:
     #  - php=${{package.full-version}}
     runtime:
       - libxml2
+      - ${{package.name}}-config
 
 environment:
   contents:
@@ -140,14 +141,9 @@ pipeline:
     runs: |
       cd $HOME/php-${{vars.mangled-package-version}}
       INSTALL_ROOT=${{targets.destdir}} DESTDIR=${{targets.destdir}} make install
+      mkdir -p "${{targets.destdir}}/bin" && ln -s /usr/bin/php "${{targets.destdir}}/bin/php"
 
   - uses: strip
-
-  - name: Copy configuration and set up alias on /bin
-    runs: |
-      mkdir -p "${{targets.destdir}}/etc/php/conf.d"
-      mv $HOME/php-${{vars.mangled-package-version}}/php.ini-production ${{targets.destdir}}/etc/php/php.ini
-      mkdir -p "${{targets.destdir}}/bin" && ln -s /usr/bin/php "${{targets.destdir}}/bin/php"
 
 data:
   - name: extensions
@@ -192,22 +188,44 @@ data:
       odbc: UnixODBC
 
 subpackages:
+  - name: ${{package.name}}-config
+    # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
+    # dependencies:
+    #   provides:
+    #     - php-${{range.key}}-config=${{package.full-version}}
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}/etc/php/conf.d"
+          mv $HOME/php-${{vars.mangled-package-version}}/php.ini-production ${{targets.subpkgdir}}/etc/php/php.ini
+
   - range: extensions
     name: "${{package.name}}-${{range.key}}"
     description: "The ${{range.value}} extension"
+    dependencies:
+      runtime:
+        - "${{package.name}}-${{range.key}}-config"
     # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
-    #dependencies:
     #  provides:
     #    - php-${{range.key}}=${{package.full-version}}
     pipeline:
       - runs: |
           export EXTENSIONS_DIR=usr/lib/php/modules
-          export CONF_DIR="${{targets.subpkgdir}}/etc/php/conf.d"
-          mkdir -p "${{targets.subpkgdir}}"/$EXTENSIONS_DIR $CONF_DIR
+          mkdir -p "${{targets.subpkgdir}}"/$EXTENSIONS_DIR
           mv "${{targets.destdir}}/$EXTENSIONS_DIR/${{range.key}}.so" \
             "${{targets.subpkgdir}}/$EXTENSIONS_DIR/${{range.key}}.so"
-          prefix=
-          [ "${{range.key}}" != "opcache" ] || prefix="zend_"
+
+  - range: extensions
+    name: "${{package.name}}-${{range.key}}-config"
+    description: "The ${{range.value}} extension configuration"
+    # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
+    #dependencies:
+    #  provides:
+    #    - php-${{range.key}}-config=${{package.full-version}}
+    pipeline:
+      - runs: |
+          export CONF_DIR="${{targets.subpkgdir}}/etc/php/conf.d"
+          mkdir -p $CONF_DIR
+          prefix=[ "${{range.key}}" != "opcache" ] || prefix="zend_"
           echo "${prefix}extension=${{range.key}}.so" > $CONF_DIR/"${{range.key}}.ini"
 
   - name: ${{package.name}}-dev
@@ -223,7 +241,7 @@ subpackages:
           mv ${{targets.destdir}}/usr/bin/phpize ${{targets.subpkgdir}}/usr/bin/
           mv ${{targets.destdir}}/usr/lib ${{targets.subpkgdir}}/usr
 
-  - name: ${{package.name}}-cgi
+  - name: "${{package.name}}-cgi"
     description: PHP 8.3 CGI
     # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
     #dependencies:
@@ -234,7 +252,7 @@ subpackages:
           mkdir -p ${{targets.subpkgdir}}/usr/bin
           mv ${{targets.destdir}}/usr/bin/php-cgi ${{targets.subpkgdir}}/usr/bin/
 
-  - name: ${{package.name}}-dbg
+  - name: "${{package.name}}-dbg"
     description: Interactive PHP Debugger
     # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
     #dependencies:
@@ -245,16 +263,28 @@ subpackages:
           mkdir -p ${{targets.subpkgdir}}/usr/bin
           mv ${{targets.destdir}}/usr/bin/phpdbg ${{targets.subpkgdir}}/usr/bin/
 
-  - name: ${{package.name}}-fpm
+  - name: "${{package.name}}-fpm"
     description: PHP 8.3 FastCGI Process Manager (FPM)
-    # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
     dependencies:
+      runtime:
+        - "${{package.name}}-fpm-config"
+    # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
     #  provides:
     #    - php-fpm=${{package.full-version}}
     pipeline:
       - runs: |
-          mkdir -p ${{targets.subpkgdir}}/usr/sbin ${{targets.subpkgdir}}/etc/php/php-fpm.d
+          mkdir -p ${{targets.subpkgdir}}/usr/sbin
           mv ${{targets.destdir}}/usr/sbin/php-fpm ${{targets.subpkgdir}}/usr/sbin/
+
+  - name: ${{package.name}}-fpm-config
+    description: PHP 8.3 FastCGI Process Manager (FPM) configuration
+    # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
+    # dependencies:
+    #  provides:
+    #    - php-fpm-config=${{package.full-version}}
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/etc/php/php-fpm.d
           mv ${{targets.destdir}}/etc/php/php-fpm.conf.default ${{targets.subpkgdir}}/etc/php/php-fpm.conf
           mv ${{targets.destdir}}/etc/php/php-fpm.d/www.conf.default ${{targets.subpkgdir}}/etc/php/php-fpm.d/www.conf \
           && { \

--- a/php-8.3.yaml
+++ b/php-8.3.yaml
@@ -41,7 +41,7 @@ environment:
       - oniguruma-dev
       - openldap-dev
       - openssl-dev
-      - postgresql-15-dev
+      - postgresql-16-dev
       - readline-dev
       - sqlite-dev
       - unixodbc-dev
@@ -225,7 +225,8 @@ subpackages:
       - runs: |
           export CONF_DIR="${{targets.subpkgdir}}/etc/php/conf.d"
           mkdir -p $CONF_DIR
-          prefix=[ "${{range.key}}" != "opcache" ] || prefix="zend_"
+          prefix=
+          [ "${{range.key}}" != "opcache" ] || prefix="zend_"
           echo "${prefix}extension=${{range.key}}.so" > $CONF_DIR/"${{range.key}}.ini"
 
   - name: ${{package.name}}-dev
@@ -240,6 +241,15 @@ subpackages:
           mkdir -p ${{targets.subpkgdir}}/usr/bin
           mv ${{targets.destdir}}/usr/bin/phpize ${{targets.subpkgdir}}/usr/bin/
           mv ${{targets.destdir}}/usr/lib ${{targets.subpkgdir}}/usr
+
+  - name: ${{package.name}}-doc
+    description: PHP 8.3 documentation
+    # DO NOT USE PROVIDES UNTIL A RELEASE HAS BEEN MADE (not a release candidate)
+    #dependencies:
+    #  provides:
+    #    - php-doc=${{package.full-version}}
+    pipeline:
+      - uses: split/manpages
 
   - name: "${{package.name}}-cgi"
     description: PHP 8.3 CGI

